frontend/catalyst/device/decomposition.py

```diff
@@ -148,10 +72,12 @@ def catalyst_decomposer(op, capabilities: DeviceCapabilities):
     if alternative_decomp is not None:
         return alternative_decomp
 
-    if capabilities.to_matrix_ops.get(op.name) or (
-        op.has_matrix and isinstance(op, qml.ops.Controlled)
-    ):
-        return _decompose_to_matrix(op)
+    if capabilities.native_ops.get("QubitUnitary"):
+        # If the device supports unitary matrices, apply the relevant conversions and fallbacks.
+        if capabilities.to_matrix_ops.get(op.name) or (
+            op.has_matrix and isinstance(op, qml.ops.Controlled)
+        ):
+            return _decompose_to_matrix(op)
```

Comment:
This is minor, but if a TOML file specifies `operator.gate.matrix` but does not have `QubitUnitary` in `operator.gate.native`, we might want to raise an error?

Reply:
Agreed, that's something better done in the toml validation though, not the decomposition machinery :)